packages/base/src/tools.ts

```diff
@@ -557,20 +596,55 @@ export const loadFile = async (fileInfo: {
           return cached.file;
         }
 
+        try {
+          const response = await fetch(filepath);
+          if (!response.ok) {
+            throw new Error(`Failed to fetch GeoJSON from URL: ${filepath}`);
+          }
+          const geojson = await response.json();
+          await saveToIndexedDB(filepath, geojson);
+          return geojson;
+        } catch (error) {
+          console.warn(`Cannot load GeoJSON from ${filepath}:`, error);
+        }
+
+        // Trying through our proxy server
         try {
           const response = await fetch(
             `/jupytergis_core/proxy?url=${filepath}`
           );
           if (!response.ok) {
-            throw new Error(`Failed to fetch GeoJSON from URL: ${filepath}`);
+            throw new Error(
+              `Failed to fetch GeoJSON via internal proxy: ${filepath}`
+            );
           }
           const geojson = await response.json();
           await saveToIndexedDB(filepath, geojson);
           return geojson;
         } catch (error) {
-          console.error('Error loading remote GeoJSON:', error);
-          throw error;
+          console.warn(
+            'Cannot communicate with the JupyterGIS proxy server:',
+            error
+          );
         }
+
+        // Trying through an external proxy server
+        try {
+          const response = await fetch(`https://corsproxy.io/?url=${filepath}`);
```

Comment:
Can we take this opportunity to fix https://github.com/geojupyter/jupytergis/issues/457 and have one single function `fetchFile(url: string): any` that does this? Going through the proxy to download an outside asset can be implemented in one function that we reuse multiple times instead of copying this code.

Reply:
Sure, i'll be on it :)